docs/proposals/20220411-cluster-api-state-metrics.md

```diff
@@ -174,9 +179,30 @@ It caches the current state of the metrics using an internal cache and updates t
 
 On requests to `/metrics` the cached data gets concatenated to a single string and returned as a response.
 
-#### Reuse kube-state-metrics packages
+### Use kube-state-metrics custom resource configuration
+
+kube-state-metrics v2.5.0 introduced an experimental feature to create metrics for CRDs.
+The feature for v2.5.0 did not fulfill all requirements for the metrics of this proposal.
+Because of that improvements got contributed to kube-state-metrics to improve the feature and support more use-cases, including the metrics of this proposal.
+
+For the implementation of this proposal a proper configuration file for kube-state-metrics should be provided and kube-state-metrics should be used to expose the metrics.
 
-Cluster-api-state-metrics should re-use as many packages provided by kube-state-metrics as possible. This allows re-use of flags, configuration and extended functionality like sharding or tls configuration without additional implementation.
+### Security Model
+
+- RBAC definitions should be generated via kubebuilder annotations in future.
```

Comment:
That is a very interesting point. Although I'm not sure if it's worth building something to generate them (we need a source file somewhere for that)

Maybe writing them manually is good enough?

Reply:
Could be also done by the configuration generator.

By knowing that we have a configuration for the api resource, it also knows that it requires an rbac rule to allow access to the resource.
The config includes the same information as needed for rbac in that case, except the verbs (which are always `get`,`list`,`watch`).

Reply:
Sounds good. Can we generalize the wording a bit. Currently it states `via kubebuilder annotations`.